src/target/ext/base/single_register_access.rs

```diff
@@ -22,16 +22,18 @@ pub trait SingleRegisterAccess<Id>: Target {
     /// On single threaded targets, `tid` is set to `()` and can be ignored.
     ///
     /// Implementations should write the value of the register using target's
-    /// native byte order in the buffer `dst`.
+    /// native byte order in the buffer `buf`.
     ///
     /// If the requested register could not be accessed, an appropriate
     /// non-fatal error should be returned.
+    ///
+    /// Return the number of bytes written into `buf`.
```

Comment:
nit: move this line right after "Implementations should write the value of the register using target's native byte order in the buffer `buf`"